src/robot.coffee

```diff
@@ -247,6 +264,28 @@ class Robot
   # Returns nothing.
   # Returns before executing callback
   receive: (message, cb) ->
+    # When everything is finished (down the middleware stack and back up),
+    # pass control back to the robot
+    receiveMiddlewareDone = () ->
+      if cb?
+        process.nextTick -> cb true
```

Comment:
Seems okay to leave here for now. If there's another case that needs it in the executor stage, can refactor it then. (Would https://github.com/github/hubot/pull/1021 need it?)